include/proxsuite/proxqp/dense/preconditioner/ruiz.hpp

```diff
@@ -120,10 +132,16 @@ ruiz_scale_qp_in_place( //
 
       for (isize k = 0; k < n_eq; ++k) {
         T aux = sqrt(infty_norm(A.row(k)));
+        if (aux == T(0)) {
+          aux = T(1);
+        }
         delta(n + k) = T(1) / (aux + machine_eps);
       }
       for (isize k = 0; k < n_in; ++k) {
         T aux = sqrt(infty_norm(C.row(k)));
+        if (aux == T(0)) {
+          aux = T(1);
+        }
```

Comment:
@Bambade Why these lines?

Reply:
If someone puts a zero column in A or H, the associated coefficient of the linear part of the pre-conditioner is close to +inf (=1/(0 + eps) multiplied 10 times if max_iter= 10, since ||1-delta|| != 0 because of the inf). It makes the procedure less efficient than just putting 1 instead.